comit/src/network/protocols/orderbook/take_order.rs

```diff
@@ -0,0 +1,116 @@
+use crate::network::OrderId;
+use futures::{prelude::*, AsyncWriteExt};
+use libp2p::{
+    core::upgrade,
+    request_response::{ProtocolName, RequestResponseCodec},
+};
+use serde::Deserialize;
+use std::io;
+use tracing::debug;
+
+#[derive(Clone)]
+pub struct TakeOrderProtocol;
+
+impl ProtocolName for TakeOrderProtocol {
+    fn protocol_name(&self) -> &[u8] {
+        b"/comit/orderbook/take/1.0.0"
+    }
+}
+
+#[derive(Clone, Default)]
+pub struct TakeOrderCodec;
+
+/// The different responses we can send back as part of an announcement.
+///
+/// For now, this only includes a generic error variant in addition to the
+/// confirmation because we simply close the connection in case of an error.
+#[derive(Debug, Deserialize)]
+pub enum Response {
+    Confirmation,
+    Error,
+}
+
+#[async_trait::async_trait]
+impl RequestResponseCodec for TakeOrderCodec {
+    type Protocol = TakeOrderProtocol;
+    type Request = OrderId;
```

Comment:
Request should include a `SwapDigest`.

Reply:
Oh yeah? Swap digest is no-where in `Orderbook` or `TakeOrder` right now - do we need it?

Reply:
Well, is there any data that the parties don't share and we still need to assert that they have the same understanding of it?

If you can answer this question, then you know whether or not you need `SwapDigest` because that is what it was designed for.

Reply:
I'm shooting for having all info contained within the order so no need for a swap digest.

Reply:
Cool, well then we can delete this stuff soon (once we move cnd's API to be asset-centric and all swaps are created through the orderbook!)